### PR TITLE
DEV-1525 make md5 optional

### PR DESCRIPTION
**Pull request by RudolfDG**

### CODE REVIEW

**Review by maartends — CHANGES_REQUESTED** [review]
`import re` missing in `main.py`

```
tests/test_main.py::test_construct_essence_sidecar FAILED
...
FAILED tests/test_main.py::test_construct_essence_sidecar - NameError: name 're' is not defined
```
Screenshot from 2021-03-03 20-26-36

Also, because of time-constraints, adding extra tests can/should become a new issue.

**Review by maartends — APPROVED** [review]

File: main.py [hunk 1]
```diff
@@ -102,6 +102,10 @@ def construct_essence_sidecar(event, pid, cp_name):
     etree.SubElement(mdprops, "object_use").text = "archive_master"
     etree.SubElement(mdprops, "ie_type").text = "n/a"
 
+    # Only add md5 if valid and available from the event.
+    if re.match("^[a-fA-F0-9]{32}$", get_from_event(event, "md5")):
+        etree.SubElement(mdprops, "md5").text = get_from_event(event, "md5")
+
```
Comment (spacid) [inline main.py:108]: Perhaps move this to a reusable method as it is used twice?

File: meemoo/helpers.py [hunk 2]
```diff
@@ -54,11 +55,7 @@ def try_to_find_md5(object_metadata):
         if key in object_metadata.keys():
             log.debug(f"md5sum located in metadata-field: '{key}'")
             md5 = object_metadata[key]
-    
-    if re.match("^[a-fA-F0-9]{32}$", md5):
```
Comment (spacid) [inline meemoo/helpers.py:58]: Removing this check here results in an unused import of "re".

File: meemoo/helpers.py [hunk 3]
```diff
@@ -29,6 +29,7 @@
 # Constants
 BASE_DOMAIN = "viaa.be"
 S3_FIELDS = ["bucket", "object_key", "domain", "tenant", "user", "md5", "event_name"]
```
Comment (spacid) [inline meemoo/helpers.py:31]: Is there an easy way to make this a superset of `REQUIRED_S3_FIELDS`?

**Review by spacid — APPROVED** [review]

**Pull request merged**